### PR TITLE
Fix the template for RouterUsingActions to eliminate compilation errors in code generated using it.

### PR DESCRIPTION
**Pull request by drZoid**
1. Modified the error handler type to ActorSystem => PartialFunction[Throwable, Trailers]
2. Removed the extra materializer being passed to ServiceHandler.

This fixes following [issue](https://github.com/playframework/play-grpc/issues/292) for play-grpc version 0.8.2

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
Hi @drZoid,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[https://www.lightbend.com/contribute/cla](https://www.lightbend.com/contribute/cla)

**Comment by raboof** [comment]
The travis failure is because the copyright header check incorrectly uses the 'current year', this has since been fixed in #201. I think it'd be OK to remove the headerCheck from .travis.yml for the 0.8.x branch.

**Review by raboof — COMMENTED** [review]
Change looks good to me (but we should remove the headers check from .travis.yml to make travis happy before merging)

Would it be possible to add/adapt a test, so such compilation problems would be caught on travis in the future?

**Comment by drZoid** [comment]
@raboof - The build failed for jdk-8 but all tests passed for jdk-11. I could not make much sense of the stack trace, if you have any pointers about the reason behind that it would be greatly appreciated.

**Regarding testing**
I tried adding few tests for `RouterUsingActions` by modifying `playTestData` to contain 
` .settings(akkaGrpcCodeGeneratorSettings += "use_play_actions")` in [build.sbt](https://github.com/playframework/play-grpc/blob/0.8.x/build.sbt#L59)

However, it will _either generate the traditional router or the one using Actions_ and therefore this will disrupt the existing tests.

In order to get around that we can add another module `playTestDataUsingActions`. Just running a compile on it should catch this bug. However, if we want to add proper tests for this feature including basic tests for play filters and ActionBuilders then we would have to add a separate testing module too.

Does this seem like an acceptable strategy for testing?

**Comment by drZoid** [comment]
Update - I added a module for generating test data with `RouterUsingActions` to add few unit tests. On running that I realized what the real issue is.

The code in this branch does not have a bug. It uses `akka-grpc` version `0.7.3` and the generated routers compile correctly with that version of akka-grpc.

However, I am trying to use this library with `akka-grpc` `0.8.4` and that's where this bug gets introduced and generated code with `RoutersUsingActions` stops compiling.

Looking at the commits in master, it seems to me that this bug gets introduced with the upgrade to akka-grpc 0.8.4 [here](https://github.com/playframework/play-grpc/pull/238).

Since the 0.9.0 release directly jumps to akka 1.0.0, I guess what I really want is to backport all the changes until this commit https://github.com/playframework/play-grpc/pull/238 to 0.8.x branch and release a new version compatible with akka-grpc 0.8.4 with the fix applied.

do you think this is an acceptable strategy to release a version of play-grpc compatible with akka-grpc 0.8.4?

**Comment by raboof** [comment]
> The code in this branch (0.8.x) does not have a bug. It uses akka-grpc version 0.7.3

Huh, that is wrong: that means the `0.8.x` branch is still behind the `0.8.2` tag, which of course makes no sense.

> what I really want is to backport all the changes until this commit #238 to 0.8.x branch and release a new version compatible with akka-grpc 0.8.4 with the fix applied.

How about:
* I move the 0.8.x branch up (to f3834227f916b1b255fccaba59814832a11d59c0?)
* you PR the fix on that branch
* we release 0.8.3 from that updated 0.8.x branch

(it would be even nicer if we could fix this in master, but you mentioned that appears to have another issue for you apart from this one - I'm of course curious what that other issue is ;) )

**Comment by drZoid** [comment]
> How about:
> 
>     * I move the 0.8.x branch up (to [f383422](https://github.com/playframework/play-grpc/commit/f3834227f916b1b255fccaba59814832a11d59c0)?)
> 
>     * you PR the fix on that branch
> 
>     * we release 0.8.3 from that updated 0.8.x branch
That's perfect. I am also adding a few unit tests for this feature.
 
> (it would be even nicer if we could fix this in master, but you mentioned that appears to have another issue for you apart from this one - I'm of course curious what that other issue is ;) )
My first attempt was to apply the template fix on the latest master but I ran into another compilation error and it involved taking a closer look at akka-grpc's api changes in the releases after 0.8.4. I will investigate further and send an update about this as soon as I finish the tests :)

**Comment by raboof** [comment]
> That's perfect

:+1: updated the 0.8.x branch

> I will investigate further and send an update about this as soon as I finish the tests :)

:+1:

File: build.sbt [hunk 1]
```diff
@@ -92,7 +112,7 @@ val playGenerators = Project("play-grpc-generators", file("play-generators"))
   )
 
 val playTestkit = Project("play-grpc-testkit", file("play-testkit"))
-  .dependsOn(playTestdata % "test")
+  .dependsOn(playTestdata % "test", playActionsTestData % "test")
```
Comment (drZoid) [inline build.sbt:115]: helloworld proto in `playActionsTestData` uses a different package from `playTestdata` therefore they can be combined into the same `playTestkit` module.

File: play-scalatest/src/test/scala/play/grpc/scalatest/PlayActionsScalaTestSpec.scala [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  }
+
+  implicit def ws: WSClient = app.injector.instanceOf(classOf[WSClient])
+
+  "A Play server bound to a gRPC router using actions" must {
+    "give a 404 when routing a non-gRPC request" in {
+      val result = wsUrl("/").get.futureValue
+      result.status must be(404) // Maybe should be a 426, see #396
+    }
+    // this test results in a 500
```
Comment (drZoid) [inline play-scalatest/src/test/scala/play/grpc/scalatest/PlayActionsScalaTestSpec.scala:43]: maybe we need to open issues for these test failures

**Review by drZoid — COMMENTED** [review]
I could have made the tests more 'dry' using scalatest behaviors if the failures would not have been there.

**Comment by drZoid** [comment]
Turns out these changes work on top of latest master too. Opened another pull request against master https://github.com/playframework/play-grpc/pull/296

**Review by raboof — APPROVED** [review]
LGTM, thanks!

**Pull request merged**